### ISSUE TITLE
Grid is offset

### ISSUE THREAD

**Issue by shirdekel** [minad/osm#5]
When I run `osm-home` I get the following:
![Screen Shot 2022-03-16 at 3 42 11 pm](https://user-images.githubusercontent.com/49865192/158518070-f59cc7c3-5937-4d28-8db5-a06a967ecec5.png)

How can I set osm to fill the buffer properly?

Setup
- M1 Pro Macbook
- macOS 12.0.1
- emacs-plus@28

**Comment by minad** [comment]
Ist this spacemacs? It seems there is some global setting which breaks osm and we should umdo that setting in the osm-mode buffer.

**Comment by shirdekel** [comment]
Yep, spacemacs. What do you recommend I do to narrow down the issue?

**Comment by minad** [comment]
Please check if `truncate-lines` is non-nil in the `osm-mode` buffer. Does Spacemacs overwrite this setting globally?

**Comment by shirdekel** [comment]
That fixed it! For some reason `truncate-lines` was nil for me.

![Screen Shot 2022-03-16 at 8 13 38 pm](https://user-images.githubusercontent.com/49865192/158556408-df8221a8-ae1b-4617-ad7d-7effffecd63a.png)

**Comment by shirdekel** [comment]
Turns out the issue was actually because I use `(global-visual-line-mode t)` in my `init.el`. What's the best way to keep that off only for osm-mode?

**Comment by lebensterben** [comment]
@shirdekel  

Try

(add-hook 'osm-mode-hook (lambda () (truncate-lines-mode -1)))

(hand-typed on my phone, so you need to check I made no typo)

**Comment by minad** [comment]
@shirdekel I'd avoid global minor modes. Most of them are too intrusive. Also if you use global modes from packages they prohibit lazy loading of the package. Personally I use only very few benign global modes and activate most of the modes specifically only in prog or text buffers. I just checked my config and I have only these global modes for example: global-hl-todo-mode, global-auto-revert-mode, global-diff-hl-mode, corfu-global-mode. Everything else is enabled on a case by case basis. In any case, if you use global modes, carefully check what they do. global-visual-line-mode seems broken to me if it is turned on even in special mode buffers as in this case. Of course osm-mode is quite special since it only displays images.

**Comment by shirdekel** [comment]
Cool so I ended up doing this instead of the global mode:

`(add-hook 'text-mode-hook (lambda () (visual-line-mode t)))`

Thank you for taking the time to help!

**Comment by dalanicolai** [comment]
FYI a `:predicate` keyword for `define-globalized-minor-mode` was introduced 2 years ago. However, this is not yet used frequently because it is not compatible with older Emacs versions.
Anyway, the hook works well of course, but you could also 'manage modes' by redefining the mode using `define-globalized-minor-mode`, and for sure it is handy to know about it in the future.

@minad `pdf-tools` uses a `pdf-view-incompatible-modes` to check for incompatible modes. Maybe you'd like to add such mechanism also (or maybe not :)
